Remove deprecated methods

colin-pm opened the pull request:
All methods marked as deprecated for the 1.0.0 release have been removed. All references to the deprecated methods in the documentation have been updated. Additionally, all testing has been updated to not use the deprecated methods.

This PR intendeds to close #866.

colin-pm commented:
Whoops, forgot to lint, will do that later today and update this change.

briantist commented:
I would like to bring this one up again:
- https://github.com/hvac/hvac/issues/758

Strangely it seems that the `create_token` method is not removed in this PR?

But more generally, have we been able to identify that all of the methods being removed do have replacements?

colin-pm commented on hvac/v1/__init__.py:
```diff
@@ -298,120 +295,6 @@ def get_policy(self, name, parse=False):
-        to_be_removed_in_version="1.0.0",
-        new_method=api.auth_methods.Token.create,
-    )
-    def create_token(
```
@briantist create_token is removed here in this PR

colin-pm commented:
> I would like to bring this one up again:
> 
> * [Does auth.token.create support orphan tokens? #758](https://github.com/hvac/hvac/issues/758)
> 
> Strangely it seems that the `create_token` method is not removed in this PR?

I left a comment where `create_token` was removed in this PR. I read through #758 and the Vault docs and see the issue. Since the behavior seems different between the create token and create orphan token API endpoints, I think that it would be okay to implement a `create_orphan_token` method to hit the other endpoint.

> 
> But more generally, have we been able to identify that all of the methods being removed do have replacements?

I inspected each method that I removed to ensure there was a replacement, and if there were any references to the deprecated method, modified the reference to use the replacement.

The only methods that did not have replacements were the AppID methods. I'm not sure it's worth creating replacements for those, since Vault has indicated that AppID is deprecated in favor of AppRole as of Vault 0.6.1. I would think that if anyone still needs the AppRole method, they can either use an older version of HVAC or contribute an engine to provide support. I don't think it should be a blocker for the 1.0.0 HVAC release.

colin-pm commented:
I just created #869 to address #758.

briantist reviewed (approved):
This looks good to me, I would just like to see #869 merged before this one
